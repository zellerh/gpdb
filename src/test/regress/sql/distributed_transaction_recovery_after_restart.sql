-- Given no superusers exist with a null rolvaliduntil value
set allow_system_table_mods to dml;
create table stored_superusers (role_name text, role_valid_until timestamp);
insert into stored_superusers (role_name, role_valid_until) select rolname, rolvaliduntil from pg_authid where rolsuper = true;
update pg_authid set rolvaliduntil = 'infinity' where rolsuper = true;

-- And a non-superuser can log in
create user testinguser with nocreatedb nosuperuser nocreaterole;
create database testinguser;
\! echo "local testinguser testinguser trust" >> $MASTER_DATA_DIRECTORY/pg_hba.conf;

-- When the master server restarts
\! pg_ctl -D $MASTER_DATA_DIRECTORY restart -w -m fast;

-- And a non-superuser happens to be the first user to connect
-- Then the connection should succeed
\! psql -U testinguser -c 'select 1;';

-- cleanup
\c
drop database testinguser;
drop user testinguser;
set allow_system_table_mods to dml;
update pg_authid set rolvaliduntil = stored_superusers.role_valid_until from stored_superusers where rolname=stored_superusers.role_name;
drop table stored_superusers;

\! sed -i 's/local testinguser testinguser trust//' $MASTER_DATA_DIRECTORY/pg_hba.conf
\! pg_ctl -D $MASTER_DATA_DIRECTORY restart -w -m fast;
