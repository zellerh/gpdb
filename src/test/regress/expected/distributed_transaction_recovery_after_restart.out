-- Given no superusers exist with a null rolvaliduntil value
set allow_system_table_mods to dml;
create table stored_superusers (role_name text, role_valid_until timestamp);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'role_name' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
insert into stored_superusers (role_name, role_valid_until) select rolname, rolvaliduntil from pg_authid where rolsuper = true;
update pg_authid set rolvaliduntil = 'infinity' where rolsuper = true;
-- And a non-superuser can log in
create user testinguser with nocreatedb nosuperuser nocreaterole;
NOTICE:  resource queue required -- using default resource queue "pg_default"
create database testinguser;
\! echo "local testinguser testinguser trust" >> $MASTER_DATA_DIRECTORY/pg_hba.conf;
-- When the master server restarts
\! pg_ctl -D $MASTER_DATA_DIRECTORY restart -w -m fast;
waiting for server to shut down.... done
server stopped
waiting for server to start.... done
server started
-- And a non-superuser happens to be the first user to connect
-- Then the connection should succeed
\! psql -U testinguser -c 'select 1;';
 ?column? 
----------
        1
(1 row)

-- cleanup
\c
drop database testinguser;
drop user testinguser;
set allow_system_table_mods to dml;
update pg_authid set rolvaliduntil = stored_superusers.role_valid_until from stored_superusers where rolname=stored_superusers.role_name;
drop table stored_superusers;
\! sed -i 's/local testinguser testinguser trust//' $MASTER_DATA_DIRECTORY/pg_hba.conf
\! pg_ctl -D $MASTER_DATA_DIRECTORY restart -w -m fast;
waiting for server to shut down.... done
server stopped
waiting for server to start.... done
server started
